scipy/signal/tests/test_signaltools.py

```diff
@@ -1115,7 +1106,7 @@ def test_invalid_dtypes(self, dtype):
     def test_none(self):
         # gh-1651, trac #1124. Ensure this does not segfault.
         with pytest.warns(UserWarning):
-            assert_raises(TypeError, signal.medfilt, None)
+            assert_raises(ValueError, signal.medfilt, None)
```

Comment:
I dislike the Warning+Error combination, aside from the use of `assert_raises`. It would be good to just raise on None directly, without warning spuriously.

Looking at the code, this is because `volume=None, kernel_size=None` ends up being transformed to
```
>>> volume
array([None], dtype=object)
>>> kernel_size
[3]
```
before being fed into

https://github.com/scipy/scipy/blob/bc41135d2dc01b5d99ac2b0a1ae28951fa42f9e5/scipy/signal/_signaltools.py#L1563-L1566

IMO, the test should roughly look as follows
```python
    def test_none(self):
        # gh-1651, trac #1124. Ensure this does not segfault.
        with pytest.raises(ValueError, match="..."):
            signal.medfilt(None)
```
This will require a small change in medfilt, probably easiest to add a branch `if volume is None: raise ...`.